airbyte-integrations/connectors/source-google-analytics-v4/unit_tests/unit_test.py

```diff
@@ -187,3 +197,25 @@ def test_check_connection_success_oauth(
     assert "refresh_token_val" in unquote(mock_auth_call.last_request.body)
     assert mock_auth_call.called
     assert mock_api_returns_valid_records.called
+
+
+def test_stream_slices_limited_by_current_date(test_config):
+    g = GoogleAnalyticsV4IncrementalObjectsBase(config=test_config)
+    stream_state = {"ga_date": "2050-05-01"}
+    slices = g.stream_slices(stream_state=stream_state)
+    current_date = pendulum.now().date().strftime("%Y-%m-%d")
```

Comment:
you better use [freezegun](https://github.com/spulec/freezegun) for testing current datetime related stuff